contracts/other/backd/HealerModule.sol

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.14;
+
+import "@yield-protocol/vault-interfaces/src/ICauldron.sol";
+import "@yield-protocol/vault-interfaces/src/ILadle.sol";
+import "@yield-protocol/utils-v2/contracts/math/WMul.sol";
+import "@yield-protocol/utils-v2/contracts/cast/CastU256I128.sol";
+import "../../LadleStorage.sol";
+import "../../utils/Giver.sol";
+
+///@title Ladle module that allows any vault to be poured to as long as it adds collateral or repays debt
+contract HealerModule is LadleStorage {
+    using WMul for uint256;
+    using CastU256I128 for uint256;
+
+    constructor(ICauldron cauldron_, IWETH9 weth_) LadleStorage(cauldron_, weth_) {}
+
+    function heal(bytes12 vaultId_, address to, int128 ink, int128 art)
+        external payable
+    {
+        require (ink >= 0, "Only add collateral");
+        require (art <= 0, "Only repay debt");
+        (bytes12 vaultId, DataTypes.Vault memory vault) = getVault(vaultId_);
+        DataTypes.Series memory series;
+        if (art != 0) series = getSeries(vault.seriesId);
+
+        int128 fee;
+        if (art > 0 && vault.ilkId != series.baseId && borrowingFee != 0)
+            fee = ((series.maturity - block.timestamp) * uint256(int256(art)).wmul(borrowingFee)).i128();
+
+        // Update accounting
+        cauldron.pour(vaultId, ink, art + fee);
+
+        // Manage collateral
+        if (ink != 0) {
+            IJoin ilkJoin = getJoin(vault.ilkId);
+            if (ink > 0) ilkJoin.join(vault.owner, uint128(ink));
+            if (ink < 0) ilkJoin.exit(to, uint128(-ink));
```

Comment:
Do we need this?

Reply:
I'm honestly not sure. The check for negative ink certainly isn't needed though since that's not a possible contingency. 
The logic is from `_pour` though.

Reply:
Removed the logic Alberto suggested.